### PR TITLE
Fixes #14543

### LINKED ISSUE #14543: default generated migration table's 'version' column data length not big enough
### What steps will reproduce the problem?
the default generated migration table with its "version" column set to varchar(180), but when we create a long alter table column migration,  it will drop some characters, such as the following one:
m170726_070258_add_reqm_source_column_temp_theme_column_baselib_config_column_deploy_date_column_pass_review_column_pass_test_column_rollback_time_column_regular_check_column_to_task
### What is the expected result?
right migration apply result
### What do you get instead?
migration want to re-apply old ones

workaround:
 issue SQL "alter table migration modify version varchar(360);"

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 5.6.30
| Operating system |Mac os 10.9.5

### PR DESCRIPTION
**Pull request by dmirogin**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14543

### CODE REVIEW

File: framework/console/controllers/BaseMigrateController.php [hunk 1]
```diff
@@ -622,6 +622,12 @@ public function actionCreate($name)
         }
 
         list($namespace, $className) = $this->generateClassName($name);
+        // Check that class name less then 251 (255 - strlen('.php'))
+        if (strlen($className) > 251) {
+            $className = substr($className, 0, 251);
+            $this->stdout('Length of the migration\'s class name was truncated', Console::FG_YELLOW);
```
Comment (cebe) [inline framework/console/controllers/BaseMigrateController.php]: Its probably better to error and let the user specify a new name. Using a truncated name does not have much value imo.

**Review by cebe — CHANGES_REQUESTED** [review]

File: framework/console/controllers/BaseMigrateController.php (hunk 1 above)
Comment (samdark) [reply framework/console/controllers/BaseMigrateController.php]: Agree.

File: framework/console/controllers/BaseMigrateController.php [hunk 2]
```diff
@@ -622,6 +622,11 @@ public function actionCreate($name)
         }
 
         list($namespace, $className) = $this->generateClassName($name);
+        // Check that class name less then 251 (255 - strlen('.php'))
+        if (strlen($className) > 251) {
```
Comment (rob006) [inline framework/console/controllers/BaseMigrateController.php]: It should be 180:

https://github.com/yiisoft/yii2/blob/980cfde0f1720a9f8516f76d37afa5a5528882ca/framework/console/controllers/MigrateController.php#L260-L263

**Review by rob006 — CHANGES_REQUESTED** [review]

**Comment by rob006** [comment]
I don't like using hardcoded limit - this should be configurable. Or better - length should be dynamically detected from migration table schema. 

This should be also fixed for `migrate/up` - I can manually create migration with too long name.

**Comment by dmirogin** [comment]
Cause base migration independent on database, i created constant.

**Comment by dmirogin** [comment]
Cause base migration independent on database, i created constant.

**Comment by rob006** [comment]
You could create method and override it in `MigrationController`.

**Review by samdark — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]

**Comment by SilverFire** [comment]
Could be merged in 2.0.13

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**